rasa/nlu/training_data/training_data.py

```diff
@@ -243,15 +239,22 @@ def _fill_response_phrases(self) -> None:
             # if response_key is None, that means the corresponding intent is not a
             # retrieval intent and hence no response text needs to be fetched.
             # If response_key is set, fetch the corresponding response text
-            if example.get(RESPONSE_KEY_ATTRIBUTE) is None:
+            if example.get(RESPONSE_KEY) is None:
                 continue
 
             # look for corresponding bot utterance
             story_lookup_intent = example.get_combined_intent_response_key()
             assistant_utterances = self.responses.get(story_lookup_intent, [])
             if assistant_utterances:
-                # selecting only first assistant utterance for now
-                example.set(RESPONSE, assistant_utterances[0].get(TEXT))
+
+                # Use the first response text as training label if needed downstream
+                for assistant_utterance in assistant_utterances:
+                    if assistant_utterance.get(TEXT) is not None:
```

Comment:
```suggestion
                    if assistant_utterance.get(TEXT):
```
I don't think we should train on `""`